### DISCUSSION TITLE
Can Perl module tests on Github action with msys2?

### DISCUSSION THREAD

**Discussion by yuki-kimoto** [msys2/MSYS2-packages#2367] [Q&A]
I'm Yuki Kimoto. I'm Perl Developer.

I write Perl module SPVM.

https://metacpan.org/pod/SPVM

SPVM can create Windows Binary with msys2.

  spvmcc -I lib -o myapp Myapp

myapp is windows binary. Myapp is SPVM module.

My question is whether Github Actions can test the installation of Perl  module on msys2.

If it can, I want to add msys2 tests to Github Actions.

**Comment by 1480c1** [comment]
You can try and see if https://github.com/msys2/setup-msys2 works for you, github actions does have their own copy of msys2, but I personally prefer this one because of package caching and @eine also maintains it

**Reply by yuki-kimoto** [reply]
Thank you. 

>github actions does have their own copy of msys2

I don't know this.

Can you tell me the difference between Github own copy of msys2 and setup-msys2.

It doesn't matter if the maintainability and the amount of code are the same.

**Reply by eine** [reply]
@yuki-kimoto, please see https://github.com/msys2/setup-msys2#context.

**Reply by yuki-kimoto** [reply]
thank you.

I will try it, maybe in a few month.

I'm now try to implement cotton browser in Windows

https://github.com/yuki-kimoto/cotton

msys2 environment is needed for cotton build
